test/gtest/adaptiveavgpool.hpp

```diff
@@ -0,0 +1,377 @@
+        auto in_strides                   = adaptiveavgpool_config.ComputeStrides(in_dim);
+        auto out_dim                      = adaptiveavgpool_config.GetOutput();
+        N                                 = in_dim[0];
+        C                                 = in_dim[1];
+        std::vector<size_t> out_dim_final = {N, C};
+        if(in_dim.size() == 3)
+        {
+            D = 1;
+            H = in_dim[2];
+            W = 1;
+
+            OD = 1;
+            OH = out_dim[0];
+            OW = 1;
+            out_dim_final.push_back(OH);
+        }
+        else if(in_dim.size() == 4)
+        {
+            D = 1;
+            H = in_dim[2];
+            W = in_dim[3];
+
+            OD = 1;
+            OH = out_dim[0];
+            OW = out_dim[1];
+            out_dim_final.push_back(OH);
+            out_dim_final.push_back(OW);
+        }
+        else if(in_dim.size() == 5)
+        {
+            D = in_dim[2];
+            H = in_dim[3];
+            W = in_dim[4];
+
+            OD = out_dim[0];
+            OH = out_dim[1];
+            OW = out_dim[2];
+            out_dim_final.push_back(OD);
+            out_dim_final.push_back(OH);
+            out_dim_final.push_back(OW);
+        }
+
+        auto gen_input_value = [](auto...) {
+            return prng::gen_A_to_B<T>(static_cast<T>(-10.0f), static_cast<T>(10.0f));
+        };
+        input = tensor<T>{in_dim, in_strides}.generate(gen_input_value);
+
+        output = tensor<T>{out_dim_final};
+        std::fill(output.begin(), output.end(), std::numeric_limits<T>::quiet_NaN());
+
+        ref_output = tensor<T>{out_dim_final};
+        std::fill(ref_output.begin(), ref_output.end(), std::numeric_limits<T>::quiet_NaN());
+
+        input_dev  = handle.Write(input.data);
+        output_dev = handle.Write(output.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+        miopenStatus_t status;
+
+        auto dims = input.desc.GetNumDims();
+        if(dims == 3)
+        {
+            cpu_adaptiveavgpool_forward_1d(input, ref_output, N, C, H, OH);
+        }
+        else if(dims == 4)
+        {
+            cpu_adaptiveavgpool_forward_2d(input, ref_output, N, C, H, W, OH, OW);
+        }
+        else if(dims == 5)
+        {
+            cpu_adaptiveavgpool_forward_3d<T>(input, ref_output, N, C, D, H, W, OD, OH, OW);
+        }
+        status = miopen::adaptiveavgpool::AdaptiveAvgPoolForward(
+            handle, input.desc, input_dev.get(), output.desc, output_dev.get());
+        fflush(stdout);
+        ASSERT_EQ(status, miopenStatusSuccess);
+
+        output.data = handle.Read<T>(output_dev, output.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
+
+        auto error = miopen::rms_range(ref_output, output);
+
+        ASSERT_EQ(miopen::range_distance(ref_output), miopen::range_distance(output));
+        EXPECT_LT(error, threshold * 10);
```

Comment:
You should print something like this
```suggestion
        EXPECT_LT(error, threshold * 10) << "Error forward Output beyond 10xthreshold : " << error << " Tolerance: " << threshold * 10;
```

Reply:
Resolved